src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/PackageMapReferenceAnalyzer.cs

```diff
@@ -36,10 +36,14 @@ public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAn
                 throw new ArgumentNullException(nameof(state));
             }
 
+            var currentTFM = project.TFM;
+
             // Get package maps as an array here so that they're only loaded once (as opposed to each iteration through the loop)
-            var packageMaps = await _packageMapProvider.GetPackageMapsAsync(token).ToArrayAsync(token).ConfigureAwait(false);
+            var allPackageMaps = await _packageMapProvider.GetPackageMapsAsync(token).ToArrayAsync(token).ConfigureAwait(false);
+            var packageMaps = currentTFM.IsFramework ? allPackageMaps.Where(x => x.NetCorePackagesWorkOnNetFx).ToArray<NuGetPackageMap>() : allPackageMaps;
```

Comment:
Do you need the call to `.ToArray()`? Can it remain an enumerable?

Reply:
This is to align with the packageMaps that is NugetPackageMap[]